fuel/datasets/cifar10.py

```diff
@@ -40,46 +33,31 @@ class CIFAR10(IndexableDataset):
 
     """
     provides_sources = ('features', 'targets')
-    folder = 'cifar10'
-    files = {
-        'train': [os.path.join('cifar-10-batches-py',
-                               'data_batch_{}'.format(i))
-                  for i in range(1, 6)],
-        'test': ['cifar-10-batches-py/test_batch']
-    }
+    filename = 'cifar10.hdf5'
 
-    def __init__(self, which_set, flatten=True, **kwargs):
+    def __init__(self, which_set, flatten=True, dtype=config.floatX,
```

Comment:
You should remove the `flatten` kwarg, since `H5PYDataset` already has one (which accepts a tuple of source names; that means you'll have to adapt your unit tests).